README.md

```diff
@@ -58,54 +62,68 @@ $ vmdk-convert testvm-flat.vmdk disk1.vmdk
+* `--num-cpus`: The number of CPUs of the OVA template. Default value is `2`.
+* `--mem-size`: The memory size in MB of the OVA template. Default value is `1024`.
+* `--firmware`: The firmare of the OVA template (`efi` or `bios`). Default value is `efi`.
+
+(these settings can also be set with the environment variables `NUM_CPUS`, `MEM_SIZE` and `FIRMWARE`)
 
-4. Run `mkova.sh` to create OVA with specific hardware version.
+Example: run `mkova.sh` to create OVA with specific hardware version:
 ```
-$ mkova.sh ova_name path_to_ovf_template disk1.vmdk
+$ mkova.sh --num-cpus 4 --mem-size 4096 --firmware bios ova_name path_to_ovf_template disk1.vmdk
 ```
 Where,
 * _ova_name_ is your OVA name without .ova suffix.
 * _dst.vmdk_ is the new vmdk file converted in step 2.
 * _path_to_ovf_template_ is the path to .ovf template file. Below .ovf templates files can be used.
-    * `ova/template.ovf` is the template for BIOS VM with hardware version 7.
-    * `ova/template-hw10.ovf` is the template for BIOS VM with hardware version 10.
-    * `ova/template-hw11-bios.ovf` is the template for BIOS VM with hardware version 11.
-    * `ova/template-hw11-uefi.ovf` is the template for EFI VM with hardware version 11.
-    * `ova/template-hw13-bios.ovf` is the template for BIOS VM with hardware version 13.
-    * `ova/template-hw13-uefi.ovf` is the template for EFI VM with hardware version 13.
-    * `ova/template-hw14-bios.ovf` is the template for BIOS VM with hardware version 14.
-    * `ova/template-hw14-uefi.ovf` is the template for EFI VM with hardware version 14.
-    * `ova/template-hw15.ovf` is the template for EFI or BIOS VM with hardware version 15.
-    * `ova/template-hw17.ovf` is the template for EFI or BIOS VM with hardware version 17.
-    * `ova/template-hw18.ovf` is the template for EFI or BIOS VM with hardware version 18.
-    * `ova/template-hw19.ovf` is the template for EFI or BIOS VM with hardware version 19.
-    * `ova/template-hw20.ovf` is the template for EFI or BIOS VM with hardware version 20.
-
-If you want to add more than 1 disk into the OVA, firstly convert all flat vmdk files, and add new converted vmdk files by following path_to_ovf_template.
-For example, below command creates an OVA with 3 disks
+    * `templates/template.ovf` is the template for BIOS VM with hardware version 7.
+    * `templates/template-hw10.ovf` is the template for BIOS VM with hardware version 10.
+    * `templates/template-hw11-bios.ovf` is the template for BIOS VM with hardware version 11.
+    * `templates/template-hw11-uefi.ovf` is the template for EFI VM with hardware version 11.
+    * `templates/template-hw13-bios.ovf` is the template for BIOS VM with hardware version 13.
+    * `templates/template-hw13-uefi.ovf` is the template for EFI VM with hardware version 13.
+    * `templates/template-hw14-bios.ovf` is the template for BIOS VM with hardware version 14.
+    * `templates/template-hw14-uefi.ovf` is the template for EFI VM with hardware version 14.
```

Comment:
Hi Oliver,
Do you think we can merge templates with different firmwares for HWv 11 ~ 14 into one like templates for HWv >= 15? I didn't make it because I am afraid users might have integrated this project into their automation workflow, and the changes would break their workflow. Since you have made lots of changes, how about making templates be more simplification?

Reply:
I prefer template to be parametrized, for example firmware is already an input, so no needs to have -bios/-efi suffixes.

Reply:
I agree, and that's also Qi's idea in another comment.

I think we should still have an option to use a specific template, in case a user has customized their own.